tests/jq.test

```diff
@@ -1814,3 +1814,35 @@ true
 tojson | fromjson
 {"a":nan}
 {"a":null}
+
```

Comment:
Can you organize tests in sections like the rest of `jq.test`?

Reply:
Not sure what you mean.  Do you mean: create new sections preceded by comments?

Reply:
AFAICT the format we are using for tests in `jq.tests` is:

`# section title`, 2 blank lines, and then a sequence of tests separated by a blank line, with optionally a subsection title.

```jq
# section1

# subsection 1.1

1
1
1

2
2
2

# subsection 1.2

3
3
3


# section 2

4
4
4
```

It is not consistent through the entire file, some earlier tests use
```jq
#
# section 1
#

# subsection 1.1

1
1
1
```

or
```jq
# section 1
1
1
1

2
2
2

# section 2
3
3
3
```

But generally thoughout the file, tests seem to be divided in sections that describe what they are testing, or what category of things they are testing, they are not just appended to the end without a title.

Also, since you are dividing those new tests with only one space from the previous ones, it looks like the are grouped with the tests related to parsing `nan` in JSON stings/inputs.

```jq
# nan is parsed as a valid NaN value from JSON

fromjson | isnan
"nan"
true

tojson | fromjson
{"a":nan}
{"a":null}

pick(.a.b.c)
null
{"a":{"b":{"c":null}}}
￼
pick(first)
[1,2]
[1]

...
```